core/lib/da_client/src/clients/gcs/mod.rs

```diff
@@ -0,0 +1,61 @@
+use std::{
+    fmt,
+    fmt::{Debug, Formatter},
+    sync::Arc,
+};
+
+use async_trait::async_trait;
+use zksync_config::ObjectStoreConfig;
+use zksync_da_layers::{
+    types::{DataAvailabilityError, DispatchResponse, InclusionData},
+    DataAvailabilityInterface,
+};
+use zksync_object_store::{ObjectStore, ObjectStoreFactory};
+use zksync_types::{pubdata_da::StorablePubdata, L1BatchNumber};
+
+pub(crate) struct GCSDAClient {
+    object_store: Arc<dyn ObjectStore>,
+}
+
+impl GCSDAClient {
+    pub async fn new(object_store_conf: ObjectStoreConfig) -> Self {
+        GCSDAClient {
+            object_store: ObjectStoreFactory::create_from_config(&object_store_conf).await,
+        }
+    }
+}
+
+#[async_trait]
+impl DataAvailabilityInterface for GCSDAClient {
+    async fn dispatch_blob(
+        &self,
+        batch_number: u32,
+        data: Vec<u8>,
+    ) -> Result<DispatchResponse, DataAvailabilityError> {
+        let key = self
+            .object_store
+            .put(L1BatchNumber(batch_number), &StorablePubdata { data })
```

Comment:
maybe we should also include the current timestamp? (or random nonce here?)

Reply:
we can, but I doubt that it's required as this client uses a separate bucket, and the batch numbers are unique

but even in case the data is being dispatched for the same batch the second time - it will just be overwritten, and there will be no error